### PR TITLE
Fix and improve TikTok TTS

### PR DESCRIPTION
**Pull request by jotonedev**
# Description

Additional improvements to the pull request of [jcolladosp](https://github.com/elebumm/RedditVideoMakerBot/pull/1252).

- Added new voices
- Added the option to automatically choose the voice (done by TikTok API)
- Added error handling
- Added typing annotation for better readability
- Also added more comments

# Issue Fixes

- TikTok TTS API

# Checklist:

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

# Any other information (e.g how to test the changes)

None

### CODE REVIEW

**Comment by jotonedev** [comment]
Yes, I tested it and it works.

**Comment by JasonLovesDoggo** [comment]
I don't see any documentation on how to get session ID

**Comment by OpenSourceSimon** [comment]
> I don't see any documentation on how to get session ID

I've made a pull request on the documentation repository

File: TTS/TikTok.py [hunk 1]
```diff
@@ -55,44 +62,100 @@
...
+
+        if self._code == 2:
+            return f"Code: {self._message}, reason: the text is too long, message: {self._message}"
+
+        if self._code == 4:
+            return f"Code: {self._message}, reason: the speaker doesn't exist, message: {self._message}"
 
-    def randomvoice(self):
-        return random.choice(self.voices["human"])
+        return f"Code: {self._message}, reason: unknown, message: {self._message}"
```
[the remark this answers is not in the document]
Comment (br0coder) [reply TTS/TikTok.py:161]: > The attribute / property for Code should be {self._message} and not {self._code} ?

I mean should {self._code} like this: 
` return f"Code: {self._code}, reason: unknown, message: {self._message}"`

And for all the the If statements above inside __str__() ?

File: TTS/TikTok.py [hunk 2]
```diff
@@ -55,44 +62,100 @@
...
+
+    def __str__(self) -> str:
+        if self._code == 1:
+            return f"Code: {self._message}, reason: probably the aid value isn't correct, message: {self._message}"
+
+        if self._code == 2:
+            return f"Code: {self._message}, reason: the text is too long, message: {self._message}"
+
+        if self._code == 4:
+            return f"Code: {self._message}, reason: the speaker doesn't exist, message: {self._message}"
```
Comment (br0coder) [inline TTS/TikTok.py:159]: `return f"Code: {self._message}, reason: the speaker doesn't exist, message: {self._message}"`

The error message should be returning the error number `f"Code: {self._code}"`  first and not the error message twice.

**Review by br0coder — CHANGES_REQUESTED** [review]
Thank you. I hope this helps!

File: TTS/TikTok.py (hunk 2 above)
Comment (OpenSourceSimon) [reply TTS/TikTok.py:159]:
```suggestion
            return f"Code: {self._code}, reason: the speaker doesn't exist, message: {self._message}"
```

**Review by OpenSourceSimon — CHANGES_REQUESTED** [review]

File: utils/.config.template.toml [hunk 3]
```diff
@@ -34,7 +34,8 @@ background_choice = { optional = true, default = "minecraft", example = "rocket-
 voice_choice = { optional = false, default = "tiktok", options = ["streamlabspolly", "tiktok", "googletranslate", "awspolly", "pyttsx", ], example = "tiktok", explanation = "The voice platform used for TTS generation. This can be left blank and you will be prompted to choose at runtime." }
 aws_polly_voice = { optional = false, default = "Matthew", example = "Matthew", explanation = "The voice used for AWS Polly" }
 streamlabs_polly_voice = { optional = false, default = "Matthew", example = "Matthew", explanation = "The voice used for Streamlabs Polly" }
-tiktok_voice = { optional = false, default = "en_us_006", example = "en_us_006", explanation = "The voice used for TikTok TTS" }
+tiktok_voice = { optional = true, example = "en_us_006", explanation = "The voice used for TikTok TTS" }
```
[the remark this answers is not in the document]
Comment (callumio) [reply utils/.config.template.toml]: Resolved via code review.

**Review by OpenSourceSimon — APPROVED** [review]
Thank you!

**Pull request merged**